### PR TITLE
electrum: 3.2.4 -> 3.3.2 plus new dependencies.

### PR DESCRIPTION
**Pull request by clefru**
###### Motivation for this change
Upstream release. https://github.com/spesmilo/electrum/releases

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by clefru** [comment]
@GrahamcOfBorg build electrum

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please read https://nixos.org/nixpkgs/manual/#python and https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

**Comment by clefru** [comment]
> Please read https://nixos.org/nixpkgs/manual/#python and https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

Please be more specific.

**Comment by dotlambda** [comment]
The descriptions, attribute names and `pname`s need to be changed, `meta.maintainers` is missing.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5193,6 +5195,8 @@ in {
 
   casttube = callPackage ../development/python-modules/casttube { };
 
+  aiorpcX = callPackage ../development/python-modules/aiorpcX { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: This should be sorted alphabetically - at least to some degree.

File: pkgs/applications/misc/electrum/default.nix [hunk 2]
```diff
@@ -14,14 +14,17 @@ in
 
 python3Packages.buildPythonApplication rec {
   name = "electrum-${version}";
```
Comment (dotlambda) [inline pkgs/applications/misc/electrum/default.nix]:
```suggestion
  pname = "electrum";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
From https://github.com/romis2012/aiohttp-socks:
> Python >= 3.5.3

From https://github.com/kyuupichan/aiorpcX:
> Python (>= 3.6)

**Comment by Mic92** [comment]
Should this version be backported or is our current stable version fine (i.e. it does not break with other clients of the network).

**Comment by dotlambda** [comment]
Yes, `disabled` is needed.

Attribute names are the ones in `python-packages.nix`. Package names should adhere to the same name normalization rules (PEP 0503).

**Comment by clefru** [comment]
@dotlambda 

> Yes, `disabled` is needed.

I have added that.

> Attribute names are the ones in `python-packages.nix`. Package names should adhere to the same name normalization rules (PEP 0503).

Wait, did you just take 3 message to tell me indirectly also via me reading a two-page PEP, that this should be aiohttp-socks instead of aiohttp_socks? I am used to a more welcoming and cooperating approach here on NixOS/nixpkgs. I like to apologize on your behalf for massively wasting my time. I hope that you are having a better day tomorrow.

@Mic92 From my side the review is complete and this is ready to be merged.

**Comment by dotlambda** [comment]
> I like to apologize on your behalf for massively wasting my time. I hope that you are having a better day tomorrow.

Think about it the other way round: Would you have read https://nixos.org/nixpkgs/manual/#python before touching Python packages, I wouldn't have had to tell you to read it.

File: pkgs/applications/misc/electrum/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/applications/misc/electrum/default.nix]: This still needs to be changed.

File: pkgs/development/python-modules/aiohttp-socks/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ lib, fetchPypi, buildPythonPackage, pythonOlder, aiohttp }:
+
+buildPythonPackage rec {
+  pname = "aiohttp_socks";
```
Comment (dotlambda) [inline pkgs/development/python-modules/aiohttp-socks/default.nix]:
```suggestion
  pname = "aiohttp-socks";
```

File: pkgs/development/python-modules/aiorpcX/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{ lib, fetchPypi, buildPythonPackage, pythonOlder, attrs }:
+
+buildPythonPackage rec {
+  pname = "aiorpcX";
```
Comment (dotlambda) [inline pkgs/development/python-modules/aiorpcX/default.nix]:
```suggestion
  pname = "aiorpcx";
```

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]:
```suggestion
  aiorpcx = callPackage ../development/python-modules/aiorpcX { };
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/aiohttp-socks/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    inherit pname version;
+    sha256 = "0473702jk66xrgpm28wbdgpnak4v0dh2qmdjw7ky7hf3lwwqkggf";
+  };
+
+  propagatedBuildInputs = [ aiohttp ];
+
+  disabled = pythonOlder "3.5.3";
+
+  meta = {
+    description = "SOCKS proxy connector for aiohttp.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/aiohttp-socks/default.nix]:
```suggestion
    description = "SOCKS proxy connector for aiohttp";
```
See https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

File: pkgs/development/python-modules/aiorpcX/default.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    inherit pname version;
+    sha256 = "1p88k15jh0d2a18pnnbfcamsqi2bxvmmhpizmdlxfdxf8vy5ggyj";
+  };
+
+  disabled = pythonOlder "3.6";
+
+  propagatedBuildInputs = [ attrs ];
+
+  meta = {
+    description = "Transport, protocol and framing-independent async RPC client and server implementation.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/aiorpcX/default.nix]:
```suggestion
    description = "Transport, protocol and framing-independent async RPC client and server implementation";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/aiohttp-socks/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/python-modules/aiohttp-socks/default.nix]: You will have to add `pname = "aiohttp_socks"` inside `fetchPypi`.

**Comment by clefru** [comment]
> > I like to apologize on your behalf for massively wasting my time. I hope that you are having a better day tomorrow.
> 
> Think about it the other way round: Would you have read https://nixos.org/nixpkgs/manual/#python before touching Python packages, I wouldn't have had to tell you to read it.

That as helpful as "Read the library of congress", when it concerns such a small change.

Thanks your suggested changes. Would you have been more clear on what was to change, I would have had the time to pick them up. Unfortunately, this PR has exceed the time that I had allocated for it.

No, in all seriousness, you need to stop waste people's time with obscure guess-my-mind nitpicks. Closing.

**Comment by dotlambda** [comment]
You do have a point regarding the Python packaging guide. That's too long if your only goal is to update some package. Sorry about that. Must have been my prejudice against anything Bitcoin-related.

However, one could at least have expected you to read https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes since you ticked that box.

**Pull request closed without merging**